lib/client/db/dbcmd/exec.go

```diff
@@ -0,0 +1,60 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package dbcmd
+
+import (
+	"context"
+	"os/exec"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/lib/defaults"
+)
+
+// GetExecCommand returns a command that execute the provided query on target
+// database using an appropriate CLI database client.
+func (c *CLICommandBuilder) GetExecCommand(_ context.Context, query string) (*exec.Cmd, error) {
+	if !c.options.noTLS || c.options.localProxyHost == "" {
+		return nil, trace.BadParameter("query execution is only supported when using an authenticated local proxy")
+	}
+
+	switch c.db.Protocol {
+	case defaults.ProtocolPostgres:
+		return c.getPostgresExecCommand(query)
+	case defaults.ProtocolMySQL:
+		return c.getMySQLExecCommand(query)
+	default:
+		return nil, trace.BadParameter("unsupported database protocol: %v", c.db)
```

Comment:
Let's try to make errors less generic.
```suggestion
		return nil, trace.NotImplemented("%q databases do not support exec commands yet", c.db)
```

Reply:
it was pointed out earlier that `tsh` will catch NotImplemented error and print an error stating server needs to be updated:
https://github.com/gravitational/teleport/blob/7648803f12b609f8f10830fb68d6a44fdad12438/tool/tsh/common/tsh.go#L1697-L1699
I've updated the error message but kept trace.BadParameter for now. we should improve the way tsh handles NotImplemented in the future.